legend-engine-xt-relationalStore-trino-execution/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/connection/ds/specifications/keys/TrinoDatasourceSpecificationKey.java

```diff
@@ -0,0 +1,142 @@
+
+import org.finos.legend.engine.plan.execution.stores.relational.connection.ds.DataSourceSpecificationKey;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.specification.TrinoSSLSpecification;
+
+import java.util.Objects;
+
+public class TrinoDatasourceSpecificationKey
+        implements DataSourceSpecificationKey
+{
+    public String host;
+    public int port;
+    public String catalog;
+    public String schema;
+    public String clientTags;
+
+    /**
+     *  SSL level properties
+     */
+    public TrinoSSLSpecification sslSpecification;
+
+
+    private String toString(TrinoSSLSpecification sslSpecification)
+    {
+        return "TrinoSSLSpecification{" +
+                "ssl=" + sslSpecification.ssl +
+                ", trustStorePathVaultReference='" + sslSpecification.trustStorePathVaultReference + '\'' +
+                ", trustStorePasswordVaultReference='" + sslSpecification.trustStorePasswordVaultReference + '\'' +
+                '}';
+    }
+
+    private String shortId(TrinoSSLSpecification sslSpecification)
+    {
+        if(sslSpecification != null) {
+            return "TrinoSSLSpecification_" +
+                    "ssl:" + sslSpecification.ssl + "_" +
+                    "trustStorePathVaultReference:'" + sslSpecification.trustStorePathVaultReference + "_" +
+                    "trustStorePasswordVaultReference:'" + sslSpecification.trustStorePasswordVaultReference;
+        }
+        return "_";
+    }
+
+    @Override
+    public String shortId()
+    {
+        return "Trino_" +
+            "host:" + host + "_" +
+            "port:" + port + "_" +
+            "catalog:" + catalog + "_" +
+            "schema:" + schema + "_" +
+            "clientTags:" + clientTags + "_" +
+            "sslSpecification:" + shortId(this.sslSpecification) + "_";
+    }
+
+    @Override
+    public String toString()
+    {
+        return "TrinoDatasourceSpecificationKey{" +
+                "host='" + host + '\'' +
+                ", port=" + port +
+                ", catalog='" + catalog + '\'' +
+                ", schema='" + schema + '\'' +
+                ", clientTags='" + clientTags + '\'' +
+                ", sslSpecification=" + toString(this.sslSpecification) +
+                '}';
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o)
+        {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass())
+        {
+            return false;
+        }
+        TrinoDatasourceSpecificationKey that = (TrinoDatasourceSpecificationKey) o;
+        return port == that.port && Objects.equals(host, that.host) && Objects.equals(catalog, that.catalog) && Objects.equals(schema, that.schema) && Objects.equals(clientTags, that.clientTags) && Objects.equals(sslSpecification, that.sslSpecification);
```

Comment:
hashCode and equals are incorrect, as sslSpecification does not has hashCode and equals internally, It might be a better idea to convert it to string and then use it within hashCode and equals

Reply:
This is fixed.